### PR TITLE
[Master] Make ballerina shell experimental

### LINKED ISSUE #41034: [Task]: Make Shell Experimental
### Description

Due to the limitation mentioned https://github.com/ballerina-platform/ballerina-lang/issues/39960, we are going to make the shell experimental feature. 

### Describe your task(s)

_No response_

### Related area

-> Shell

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by prakanth97**
## Purpose

Fixes #41034 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: ballerina-shell/modules/shell-cli/src/main/resources/command.help.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Ballerina Shell Help
+Ballerina Shell [experimental] Help
```
Comment (lochana-chathura) [inline ballerina-shell/modules/shell-cli/src/main/resources/command.help.txt:1]:
```suggestion
Ballerina Shell Help
Note: This is an experimental feature, which supports only a limited set of functionalities.
```
WDYT? We are explicitly saying it supports only a limited set of functionalities.

Comment (lochana-chathura) [reply ballerina-shell/modules/shell-cli/src/main/resources/command.help.txt:1]: We used to have similar message in openAPI and graphQL
- https://github.com/ballerina-platform/ballerina-lang/pull/36299/files#diff-b836d10652985678b3b08494257991011db3219fe77207bfe14c1dcf3bd07863L21-L23
- https://github.com/ballerina-platform/ballerina-lang/pull/36295/files#diff-9cc742e94958fe0aff9397bdb85279260669a76aae0b515a54c3d21b74416c5cL16-L17

File: ballerina-shell/modules/shell-cli/src/main/resources/command.header.txt [hunk 2]
```diff
@@ -1,2 +1,2 @@
-Welcome to Ballerina Shell REPL.
+Welcome to Ballerina Shell REPL [experimental].
```
Comment (prakanth97) [inline ballerina-shell/modules/shell-cli/src/main/resources/command.header.txt:1]: @lochana-chathura here this is sufficient Right?

File: ballerina-shell/modules/shell-cli/src/main/resources/command.help.txt (hunk 1 above)
Comment (prakanth97) [reply ballerina-shell/modules/shell-cli/src/main/resources/command.help.txt:1]: +1

**Comment by lochana-chathura** [comment]
Shall we update the `ballerina-shell.md` as well?

**Review by hasithaa — APPROVED** [review]

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-help.help [hunk 3]
```diff
@@ -47,5 +47,5 @@ COMMANDS
         asyncapi        Generate the Ballerina sources for a given AsyncAPI definition
         persist         Manage data persistence
         bindgen         Generate the Ballerina bindings for Java APIs
-        shell           Run Ballerina interactive REPL
+        shell           Run Ballerina interactive REPL [experimental]
```
Comment (lochana-chathura) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-help.help:50]:
```suggestion
        shell           Run Ballerina interactive REPL [Experimental]
```

File: ballerina-shell/README.md [hunk 4]
```diff
@@ -1,4 +1,7 @@
-# Ballerina Shell
+# Ballerina Shell [Experimental]
```
Comment (lochana-chathura) [inline ballerina-shell/README.md:1]:
```suggestion
# Ballerina Shell
```

**Review by lochana-chathura — APPROVED** [review]

**Pull request merged**